nemo/collections/nlp/modules/common/megatron/megatron_utils.py

```diff
@@ -81,19 +82,24 @@ def get_megatron_lm_model(
             for example: megatron-bert-cased
         config_dict: model configuration parameters
         config_file: path to model configuration file. Takes precedence over config_dict if both supplied.
-        checkpoint_file: path to checkpoint file.
+        checkpoint_file: path to checkpoint file or directory if using model parallel.
     '''
     config = None
     # get default config and checkpoint
     if config_file:
         with open(config_file) as f:
-            configf = json.load(f)
-            config = {
-                "hidden_size": configf['hidden-size'],
-                "num_attention_heads": configf['num-attention-heads'],
-                "num_layers": configf['num-layers'],
-                "max_position_embeddings": configf['max-seq-length'],
-            }
+            config = json.load(f)
+            # replace dashes with underscores in config keys
+            fixed_config = {}
+            for key in config.keys():
+                fixed_key = key.replace("-", "_")
+                if fixed_key == 'max_seq_length':
+                    fixed_key = 'max_position_embeddings'
+                fixed_config[fixed_key] = config[key]
+            # 'vocab_size" no longer used.
+            if 'vocab_size' in fixed_config:
+                fixed_config.pop('vocab_size')
+            config = fixed_config
```

Comment:
Why do we need these changes here? I don't remember if the configs on NGC have `-` or `_`

Reply:
The previous code was hard coding the change so I'm assuming the old configs had the dashes. I'm hoping we can move away from the .json files soon and just input config from CLI or .yaml.

Reply:
let' not move from .json for megatron as this is want released on ngc